flopy/plot/plotutil.py

```diff
@@ -2591,3 +2592,258 @@ def parse_modpath_selection_options(
         tep = ep.copy()
 
     return tep, istart, xp, yp
+
+
+def to_mp7_pathlines(
+    data: Union[np.recarray, pd.DataFrame]
+) -> Union[np.recarray, pd.DataFrame]:
+    """
+    Convert MODFLOW 6 PRT pathline data to MODPATH 7 pathline format.
+
+    Parameters
+    ----------
+    data : np.recarray or pd.DataFrame
+        MODFLOW 6 PRT pathline data
+
+    Returns
+    -------
+    np.recarray or pd.DataFrame (consistent with input type)
+    """
+
+    # define expected fields
+    mp7_names = ["x", "y", "z", "time", "k", "particleid"]
+    prt_names = [
+        "x",
+        "y",
+        "z",
+        "t",
+        "trelease",
+        "imdl",
+        "iprp",
+        "irpt",
+        "ilay",
+    ]
+
+    # determine return type
+    ret_type = type(data)
+
+    # convert to dataframe if needed
+    if not isinstance(data, pd.DataFrame):
+        data = pd.DataFrame(data)
+
+    # check format
+    dt = data.dtypes
+    if not (
+        all(n in dt for n in mp7_names) or all(n in dt for n in prt_names)
+    ):
+        raise ValueError(
+            "Pathline data must contain all of the following columns: "
+            f"{mp7_names} for MODPATH 7, or {prt_names} for MODFLOW 6 PRT"
+        )
+
+    # return early if no need to convert
+    if "t" not in dt:
+        return (
+            data if ret_type == pd.DataFrame else data.to_records(index=False)
+        )
+
+    # assign a unique particle index column incrementing an integer
+    # for each unique combination of irpt, iprp, imdl, and trelease
+    data = data.sort_values(["imdl", "iprp", "irpt", "trelease"])
+    particles = data.groupby(["imdl", "iprp", "irpt", "trelease"])
+    seqn_key = "sequencenumber"
+    data[seqn_key] = particles.ngroup()
```

Comment:
This just mimics a modpath sequencenumber &mdash; can't guarantee particles are indexed in the same order so this should be ignored when comparing prt and modpath results

It's possible to preserve particle identity by running modpath first, then setting prt release point boundnames to modpath sequencenumbers. Could maybe add a switch `boundname_as_sequencenumber=False` to try parse an int and set sequencenumber accordingly